code/game/objects/effects/portals.dm

```diff
@@ -24,13 +24,13 @@
 	src.creator = creator
 	var/area/A = target.loc
 	if(A.noteleport) // No point in persisting if the target is unreachable.
+		src.visible_message("<span class='danger'>The [src] suddenly loses power and closes! The target destination is under the effects of a teleportation jammer!<span>")
 		qdel(src)
 		return
 	for(var/mob/M in src.loc)
 		src.teleport(M)
 	if(lifespan > 0)
-		spawn(lifespan)
-			qdel(src)
+		addtimer(src, "Destroy", lifespan)

```

Comment:
Can't call ```qdel()``` here, that'd be ```thing.qdel()``` due to the way timers work, can't do that.

Should definitely be using ```qdel()``` instead though, even if it means an extra proc definition on this type that just ```qdel()```s it.

eg:
```
/portal/proc/delete()
   qdel(src)
```